File: ruby/gmsec4/dox/ConfigPair.rb
# Copyright 2007-2021 United States Government as represented by the
# Administrator of The National Aeronautics and Space Administration.
# No copyright is claimed in the United States under Title 17, U.S. Code.
# All Rights Reserved.

# This file is used for generating Ruby documentation; it only contains
# stubs of constructors and methods associated with the class.


module Libgmsec_ruby #:nodoc: don't document this

	# This class wraps a name and value pair from a Config object.
	#
	# ==== Example Usage
	#
	#    cp = Libgmsec_ruby::ConfigPair.new
	#
	#    has_pair = config.get_first(cp)
	#
	#    while has_pair do
	#        puts("Name = #{cp.get_name}, Value = #{cp.get_value}")
	#
	#        has_pair = config.get_next(cp)
	#    end
	#
	# ==== See also
	# Config
	#
	class ConfigPair
		##
		# Returns the name associated with the configuration entry.
		#
		# ==== Returns
		#
		# A string containing the name of the configuration entry.
		#
		def get_name
		end


		##
		# Returns the value associated with the configuration entry.
		#
		# ==== Returns
		#
		# A string containing the value of the configuration entry.
		#
		def get_value
		end
	end
end

File: csharp/gmsec4/dox/ConfigPair.cs
/*
 * Copyright 2007-2021 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */

/* This file is used for generating Doxygen documentation; it only contains
   stubs of constructors and methods associated with the class.
*/


namespace GMSEC.API {

/// <summary>Container to access individual name/value pairs that are stored in a Config object.</summary>
///
/// <seealso cref="Config.GetFirst"/>
/// <seealso cref="Config.GetNext"/>

public class ConfigPair : global::System.IDisposable {

  /// <summary>Returns the name of the Config entry.</summary>
  public string GetName() {
    string ret = GmsecPINVOKE.ConfigPair_GetName(swigCPtr);
    if (GmsecPINVOKE.SWIGPendingException.Pending) throw GmsecPINVOKE.SWIGPendingException.Retrieve();
    return ret;
  }

  /// <summary>Returns the value of the Config entry.</summary>
  public string GetValue() {
    string ret = GmsecPINVOKE.ConfigPair_GetValue(swigCPtr);
    if (GmsecPINVOKE.SWIGPendingException.Pending) throw GmsecPINVOKE.SWIGPendingException.Retrieve();
    return ret;
  }

}

}
